completion.bash

```diff
@@ -78,6 +96,13 @@ _charmcraft()
                     ;;
             esac
             ;;
+        "upload")
+            case "$prev" in
+                --charm-file)
+                    _filedir charm
+                    ;;
+            esac
+            ;;
```

Comment:
also `release`'s `--revision` should autocomplete, n'eh?

Reply:
As we discussed, let's not hit the Store to autocomplete.